### PR TITLE
Feat/52 develop cart functionality to manage cart state

### PR DESCRIPTION
**Pull request by mbunwe-victor**

### CODE REVIEW

**Review by Valsuh45 — APPROVED** [review]

File: src/components/cart/cart-display.tsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+                >
+                  Update
+                </Button>
+                <Button
+                  color="secondary"
+                  onClick={() => handleRemoveItem(item.id)}
+                >
+                  Remove
+                </Button>
+              </td>
```
Comment (stephane-segning) [inline src/components/cart/cart-display.tsx:88]: Extract this to a sub-component.

File: src/components/cart/cart-display.tsx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+              <td className="border border-gray-300 p-2">{item.productId}</td>
+              <td className="border border-gray-300 p-2">${item.price.toFixed(2)}</td>
+              <td className="border border-gray-300 p-2">
+                <input
+                  type="number"
+                  min="1"
+                  value={newQuantity[item.id] || item.quantity}
+                  onChange={(e) => handleQuantityChange(item.id, parseInt(e.target.value))}
+                  className="border border-gray-300 p-2 rounded w-20"
+                />
```
Comment (stephane-segning) [inline src/components/cart/cart-display.tsx:72]: Extract this to a sub-component.

Comment (stephane-segning) [reply src/components/cart/cart-display.tsx:72]: That is, create a component and reuse it here

File: src/components/cart/cart-display.tsx (hunk 1 above)
Comment (stephane-segning) [reply src/components/cart/cart-display.tsx:88]: That is, create a component and reuse it here

**Review by stephane-segning — CHANGES_REQUESTED** [review]

File: src/components/cart/cart-display.tsx (hunk 2 above)
Comment (bryandino673) [reply src/components/cart/cart-display.tsx]: Sir we are done with the modifications , and the sub components have been created

File: src/components/cart/cart-row.tsx
Comment (stephane-segning) [inline src/components/cart/cart-row.tsx:1]: This component is dumb. Please, create a storybook for it.

File: src/components/cart/cart-display.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    }));
+  };
+
+  return (
+    <div className="container mx-auto p-4">
+      <h2 className="text-2xl font-bold mb-4">Your Cart</h2>
+      <table className="w-full border-collapse border border-gray-300">
+        <thead>
+          <tr>
+            <th className="border border-gray-300 p-2">Product</th>
```
Comment (stephane-segning) [inline src/components/cart/cart-display.tsx:57]: Create a table-th dumb component

File: src/components/cart/cart-display.tsx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+      onRemoveItem={handleRemoveItem}
+    />
+  ))}
+</tbody>
+      </table>
+      <div className="text-right mt-4">
+        <h3 className="text-xl font-bold">Subtotal: ${subtotal.toFixed(2)}</h3>
+      </div>
+      <div className="flex justify-between items-center mt-6">
+        <Button color="accent" onClick={handleClearCart}>
```
Comment (stephane-segning) [inline src/components/cart/cart-display.tsx:81]: This is a dangerous action and should be indicated. The easiest way is but using a custom color code, typically a red boutton with a trash icon.

**Review by stephane-segning — CHANGES_REQUESTED** [review]
The class "border-gray-300" depends on the color "gray" which is not part of the colors of our design system. Please, use one which is part of our design system, or create one if not present. Ask your team lead (@stephane-segning) or designer (@mbunwe-victor, @bryandino673) for more info

File: src/components/cart/cart-display.tsx (hunk 4 above)
Comment (bryandino673) [reply src/components/cart/cart-display.tsx]: The files have been updated  @stephane-segning @kouamschekina

**Review by kouamschekina — APPROVED** [review]

File: src/components/cart/cart-display.test.tsx [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+      cart: {
+        items: [
+          { id: '1', product: { name: 'Product A' }, price: 10, quantity: 2 },
+          { id: '2', product: { name: 'Product B' }, price: 20, quantity: 1 },
+        ],
+      },
+      removeFromCart: jest.fn(),
+      clearCart: jest.fn(),
+      updateCartItemQuantity: jest.fn(),
+    });
```
Comment (stephane-segning) [inline src/components/cart/cart-display.test.tsx:53]: Nice one

File: src/components/cart/quantity-input.tsx [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+}
+
+const QuantityInput: React.FC<QuantityInputProps> = ({ value, onChange }) => {
+  return (
+    <input
+      type="number"
+      min="1"
+      value={value}
+      onChange={(e) => onChange(parseInt(e.target.value, 10))}
+      className="border border-gray-300 p-2 rounded w-20"
```
Comment (stephane-segning) [inline src/components/cart/quantity-input.tsx:15]: Use neutral color instead of gray

**Review by stephane-segning — CHANGES_REQUESTED** [review]
Good job! Next is this single last fix

File: src/components/cart/quantity-input.tsx (hunk 6 above)
Comment (bryandino673) [reply src/components/cart/quantity-input.tsx]: Done sir @stephane-segning

**Review by stephane-segning — APPROVED** [review]

**Pull request merged**